### ISSUE TITLE
undefined variable: baseDir

### ISSUE THREAD

**Issue by CodeBlastr** [elboletaire/twbs-cake-plugin#8]
I've tried installing the command line "composer require ...", and by just manually adding it to my composer.json file and running "composer update".  Either way I get the same error : https://www.evernote.com/shard/s13/sh/2f86eadb-0f01-4463-9201-cc54a8bd50f5/833c05c016fc09a7a8d758df4f783157

I'm using 
 - CakePHP 3.2
 - PHP 7.0
 - Mac
 - Nginx
 - Postgresql

**Comment by CodeBlastr** [comment]
Don't know how I fixed it (I uninstalled, reinstalled, removed some bootstrap plugin loading, put it back in, ran composer update a few times along the way, and it just started working... weird)

**Comment by elboletaire** [comment]
:+1:

**Comment by guyandy** [comment]
I am running into this problem. It appears that during the installation, the /vendor/cakephp-plugins.php file changes to this:
```
$baseDir = dirname(dirname($baseDir . '/vendor/cakephp/plugin-installer/src/Installer/PluginInstaller.php'));
return [
    'plugins' => [
        'Bake' => $baseDir . '/vendor/cakephp/bake/',
        'BootstrapUI' => $baseDir . '/vendor/friendsofcake/bootstrap-ui/',
        'DebugKit' => $baseDir . '/vendor/cakephp/debug_kit/',
        'Migrations' => $baseDir . '/vendor/cakephp/migrations/',
        'Less' => $baseDir . '/vendor/elboletaire/less-cake-plugin/',

    ]
];
```

You can see how $baseDir is defined by itself and a path to PluginInstaller.php??

I believe $baseDir should be defined like this:
```$baseDir = dirname(dirname(__FILE__));```

**Comment by elboletaire** [comment]
Which version of `twbs-cake-plugin` are you using?

BTW, seems more like a cake bug.. the `/vendor/cakephp-plugins.php` file is generated by the postInstaller shell method.

**Comment by guyandy** [comment]
I ran

composer require elboletaire/twbs-cake-plugin

from the command line, so I suppose it got the latest version.

**Comment by elboletaire** [comment]
And which version of CakePHP are you using?

**Comment by guyandy** [comment]
3.2.7

**Comment by elboletaire** [comment]
I can confirm it. I'll take a look when I can.

PR are welcome.

**Comment by guyandy** [comment]
okay, thank you

**Comment by andreakappa** [comment]
Hi all, 
I have this problem too, i tried a dirty trick ( a shell copying continuously a modified version of the cakephp-vendors file). Composer now ends correctly, but the plugin is not being included (No helpers, components etc.)

**Comment by elboletaire** [comment]
I'm trying to find the issue here. BTW I've found a workaround:

Instead of requiring the plugin from the console (using `composer require`) add the plugin to your composer.json file and run `composer update`.

Doing it this way seems to work as expected (sorry, I still don't know why...).

**Comment by elboletaire** [comment]
Here's another workaround: remove the entire `vendor` plugin and then you'll be able to run `composer require elboletaire/twbs-cake-plugin`

Still searching for the issue itself.. this is really weird.

**Comment by elboletaire** [comment]
I've finally created an issue on cakephp/cakephp as I've been able to reproduce this issue with other plugins, specifically installing more than one at once from `friendsofcake` repository.

https://github.com/cakephp/cakephp/issues/8714

**Comment by elboletaire** [comment]
Confirmed that this is an issue from the `cakephp/plugin-installer` package.

The issue has been moved to https://github.com/cakephp/plugin-installer/issues/30

**Comment by elboletaire** [comment]
Fixed on [`plugin-installer` 0.0.15](https://github.com/cakephp/plugin-installer/releases/tag/0.0.15)

**Comment by andreakappa** [comment]
Updating plugin-installer to 0.0.15 actually makes it install correctly. 
Thank you very much!
